### PR TITLE
support attribute.style and attribute.styles

### PR DESCRIPTION
**Pull request by 0xda157**

### CODE REVIEW

**Review by lpil — COMMENTED** [review]
Never use pop_grapheme for parsing strings please, it is very slow! Instead use `string.split_once` and optionally the `splitter` package 🙏 

Could we also get some tests for when the CSS includes strings and extra whitespace please 🙏

**Comment by 0xda157** [comment]
> Never use pop_grapheme for parsing strings please, it is very slow! Instead use `string.split_once` and optionally the `splitter` package 🙏

Oh! I did not know this. I've updated the code to use `string.split_once`

> Could we also get some tests for when the CSS includes strings and extra whitespace please 🙏

I've added these tests.

File: src/html_lustre_converter.gleam [hunk 1]
```diff
@@ -79,6 +79,30 @@ fn print_string(t: String) -> String {
   "\"" <> string <> "\""
 }
 
+/// acc must always be `"["`
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: Remove this please, it's not documentation.

File: src/html_lustre_converter.gleam [hunk 2]
```diff
@@ -428,6 +452,8 @@ fn print_children_loop(
 
 fn print_attribute(attribute: #(String, String), mode: OutputMode) -> Document {
   case attribute.0 {
+    "style" -> parse_style_attribute(attribute.1)
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: This function says it parses the attribute, but it should be called `print_style_attribute` to match the rest of the functions.

File: src/html_lustre_converter.gleam [hunk 3]
```diff
@@ -664,6 +690,52 @@ fn print_attribute(attribute: #(String, String), mode: OutputMode) -> Document {
   }
 }
 
+fn parse_style_attribute(input: String) -> Document {
+  case parse_style_attribute_loop(input, []) {
+    Ok([]) -> doc.from_string("attribute.style(\"\", \"\")")
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: If there is no valid style attribute then drop it entirely please.

File: src/html_lustre_converter.gleam [hunk 4]
```diff
@@ -664,6 +690,52 @@ fn print_attribute(attribute: #(String, String), mode: OutputMode) -> Document {
...
+        <> ", "
+        <> print_string(value)
+        <> ")",
+      )
+    Ok(many) -> {
+      doc.from_string(
+        "attribute.styles(" <> print_pair_list(list.reverse(many), "[") <> ")",
+      )
+    }
+    // we don't know how to handle this, so we just pass the unaltered text to lustre
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: Start comments with capital letters and wrap long lines please

File: src/html_lustre_converter.gleam [hunk 5]
```diff
@@ -664,6 +690,52 @@ fn print_attribute(attribute: #(String, String), mode: OutputMode) -> Document {
...
+      doc.from_string(
+        "attribute.style("
+        <> print_string(name)
+        <> ", "
+        <> print_string(value)
+        <> ")",
+      )
+    Ok(many) -> {
+      doc.from_string(
+        "attribute.styles(" <> print_pair_list(list.reverse(many), "[") <> ")",
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: Use the `doc` module to build up the output, do not concatenate strings 🙏

File: src/html_lustre_converter.gleam [hunk 6]
```diff
@@ -664,6 +690,52 @@ fn print_attribute(attribute: #(String, String), mode: OutputMode) -> Document {
...
+    Error(Nil) ->
+      case string.trim(input) {
+        "" -> Ok(acc)
+        _ -> Error(Nil)
+      }
+    Ok(#(property, rest)) ->
+      case string.split_once(rest, on: ";") {
+        // the rest of the input is the value
+        Error(Nil) -> Ok([#(string.trim(property), string.trim(rest)), ..acc])
+        // there is another declaration
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: Start comments with capital letter please.
